### ISSUE TITLE
Using MQTT to set limit or restart inverter only works for default topic "inverter" and not for longer topics

### ISSUE THREAD

**Issue by anorak-47** [lumapu/ahoy#889]
### Platform

ESP8266

### Assembly

I did the assebly by myself

### nRF24L01+ Module

nRF24L01+ plus

### Antenna

external antenna

### Power Stabilization

Elko (~100uF)

### Connection picture

- [ ] I will attach/upload an Image of my wiring

### Version

0.6.0

### Github Hash

12a3d50

### Build & Flash Method

ESP Tools (flash)

### Setup

MQTT brocker was added.
Topic is "holog/client/pv/iv/hoy"

### Debug Serial Log output

```shell
09:23:02 I: MQTT got topic: holog/client/pv/iv/hoy/ctrl/limit/0
09:23:02 I: json: {"val":75,"path":"client","cmd":"pv","id":0}
```

### Error description

Setting limit or restarting inverter using MQTT does not work if using a TOPIC prefix path longer than one element.

Affected commands:
<TOPIC>/ctrl/limit/<INVERTER_ID>
<TOPIC>/ctrl/limit/<INVERTER_ID>

My topic is holog/client/pv/iv/hoy. 
Publishing "holog/client/pv/iv/hoy/ctrl/limit/0 75" results in the following JSON output (I compiled my own version with this debug output enabled): 

09:23:02 I: MQTT got topic: holog/client/pv/iv/hoy/ctrl/limit/0
09:23:02 I: json: {"val":75,"path":"client","cmd":"pv","id":0}

As you can see, the "cmd" is wrong.
Code seems to work only with the default topic "inverter".

**Comment by anorak-47** [comment]
Quick fix in pubMqtt.h:
Change line 

const char *p = topic;

to

const char *p = topic + strlen(mCfgMqtt->topic);

**Comment by juepi** [comment]
Seems this issue is still unfixed in the current development release (0.7.4), does the mentioned quick fix have any side effects?

**Comment by mslookup** [comment]
I closed my issue #1004 because the contents are nearly the same in this one here...

The only additional information i can add is that indeed other topics than the default topic "inverter" are working, but as soon as you start nesting the prefix, it stops working. have tried it out by myself, see my linked issue.

@anorak-47s quick fix is also working for me.
